website/blog/2023-05-04-generating-dynamic-docs.md

```diff
@@ -0,0 +1,368 @@
+
+```
+13:16:21  Running with dbt=1.3.1
+13:16:27  version: 2
+
+models:
+  - name: activity_based_interest_activated
+    description: ""
+    columns:
+      - name: id
+        description: ""
+
+      - name: user_id
+        description: ""
+
+... (truncated for example purposes)
+
+  - name: set_inactive_interest_rate_after_july_1st_in_bec_for_user
+    description: ""
+    columns:
+      - name: id
+        description: ""
+
+      - name: user_id
+        description: ""
+
+      - name: start_date
+        description: ""
+
+      - name: event_time
+        description: ""
+
+      - name: event_day
+        description: ""
+
+  - name: set_inactive_interest_rate_from_july_1st_in_bec_for_user
+    description: ""
+    columns:
+      - name: id
+        description: ""
+
+      - name: user_id
+        description: ""
+
+      - name: event_time
+        description: ""
+
+      - name: event_day
+        description: ""
+```
+
+### Get model names programmatically
+
+In order to not have to manually write all of the model names, we can programmatically gather names of relevant models:
+
+```
+$ dbt ls -m models/core/activity_based_interest --output name | xargs -I{} echo -n ' "{}",'
+ "activity_based_interest_activated", "activity_based_interest_deactivated", "activity_based_interest_updated", "downgrade_interest_level_for_user", "f_activity_based_interest", "set_inactive_interest_rate_after_july_1st_in_bec_for_user", "set_inactive_interest_rate_from_july_1st_in_bec_for_user", "set_interest_levels_from_june_1st_in_bec_for_user",%
+ ```
+
+1. `dbt ls -m models/core/activity_based_interest`: This command lists all dbt models in the models/core/activity_based_interest directory.
+1. `--output name`: This option filters the output to only show the name of each model, rather than the context + model name.
+1. `| xargs -I{} echo -n ' "{}",'`: This pipe sends the output of the previous command to `xargs`, which runs the echo command on each line of output. 
+    - `-I{}` specifies that `{}` should be replaced with the model name
+    - The `echo` command then formats the model name by wrapping it in double quotes and appending a comma and a space: `"model", "name",` 
+    - The `-n` option for `echo` removes the trailing newline
+
+The output (⚠️ except the last two characters `,%` ) can then be copy-pasted into the following:
+
+```
+dbt run-operation generate_model_yaml --args '{"model_names": [ReplaceWithYourOutputFromPreviousCommand]}'
+```
+
+Which in turn can be copy-pasted into a new `.yml` file. In our example, we write it to `_activity_based_interest.yml`.
+
+## Create docs blocks for the new columns
+
+[Docs blocks](https://docs.getdbt.com/docs/collaborate/documentation#using-docs-blocks) can be utilized to write more DRY and robust documentation. To use docs blocks normally, we could manually update our folder structure, the `.md` file, and the `.yml` file. Your file structure would look a bit like this:
```

Comment:
>To use docs blocks normally, we could manually update our folder structure

This sounds like you're gearing up to describe what to do instead, but you want them to do this right? You should switch to an imperative voice (e.g. "To use docs blocks, update your folder structure to contain an `.md` file.")